java_socketio_chatroom_client/src/main/java/io/github/marcodiri/java_socketio_chatroom_client/ChatroomClient.java

```diff
@@ -0,0 +1,55 @@
+package io.github.marcodiri.java_socketio_chatroom_client;
+
+import java.net.URI;
+import org.json.JSONObject;
+
+import io.github.marcodiri.java_socketio_chatroom_client.model.Message;
+import io.github.marcodiri.java_socketio_chatroom_client.view.ClientView;
+import io.socket.client.IO;
+import io.socket.client.IO.Options;
+import io.socket.client.Socket;
+
+public class ChatroomClient {
+
+	private final Socket socket;
+	
+	private ClientView view;
+
+	public ChatroomClient(URI uri, Options options, ClientView view) {
+		this.socket = IO.socket(uri, options);
+		this.view = view;
+	}
+
+	public Socket getSocket() {
+		return socket;
+	}
+
+	public boolean isConnected() {
+		return socket.connected();
+	}
+
+	public void connect() {
+		socket.on(Socket.EVENT_CONNECT, objects -> {
+			socket.emit("join");
+			handleMessage();
+		});
+		socket.connect();
+	}
+
+	public void disconnect() {
```

Comment:
- [ ] Remember to emit _leave_ event when disconnect

- [ ] We need to verify that the client is connected before disconnecting it and leaving the room

Reply:
- I think it's the server responsibility to remove the client from the room when it disconnects.
- [`socket.disconnect()`](https://github.com/socketio/socket.io-client-java/blob/ad23cfcca6c04dc3a0a0749954c773bb0300bbfb/src/main/java/io/socket/client/Socket.java#L420) already checks if `socket.connected()`. I think it's safe to call `disconnect()` directly.